grails-app/controllers/org/pih/warehouse/picklist/PicklistController.groovy

```diff
@@ -82,4 +89,83 @@ class PicklistController {
         println location
         render(template: "/picklist/print", model: [requisition: requisition, picklist: picklist, location: location, order: params.order])
     }
+
+    def exportPicklistItems() {
+        String format = params.get("format", "csv")
+
+        List<PickPageItem> pickPageItems = stockMovementService.getPickPageItems(params.id, null, null )
+        List<PicklistItem> picklistItems = pickPageItems.picklistItems?.flatten()
+
+        List lineItems = picklistItems.collect {
+            [
+                    id: it?.requisitionItem?.id,
+                    code: it?.requisitionItem?.product?.productCode,
+                    name: it?.requisitionItem?.product?.name,
+                    lot: it?.inventoryItem?.lotNumber,
+                    expiration: it.inventoryItem.expirationDate?.format(Constants.EXPIRATION_DATE_FORMAT),
+                    binLocation: it?.binLocation?.name,
+                    quantity: it?.quantity,
+            ]
+        }
+
+        String fileName = "PickListItems\$-${params.id}"
+
+        switch (format) {
```

Comment:
This switch here and in the next method look like could be moved to one method, or sth.

Reply:
I think we can repeat ourselves at the moment until we have a design pattern to deal with exporters and importers